### PR TITLE
Improve WebGL backend

### LINKED ISSUE #615: Implement a server-side GLIR cache
This is backend-independent. The GL context needs to store a list of (almost) all GLIR commands emitted since initialization. Unnecessary GLIR commands (old commands superseded by new commands) are automatically discarded (pruning), so that this list should never be very long.

* [ ] implement the cache (easy)
* [ ] dynamically prune old GLIR commands that are superseded by new ones: `DRAW`, `UNIFORM`, `FUNC` (slightly less straightforward)
* completely get rid of `on_initialize()` (can be done in another PR)
* fix the `ipynb_webgl` backend so that the latest state is automatically reinitialized when the context is lost (can be done in another PR)

### PR DESCRIPTION
**Pull request by rossant**
Done:

* Improved JavaScript loading, using [browserify](http://browserify.org/) instead of requireJS
* Better handling of removed widgets
* VisPy timers are automatically stopped when a widget is removed

Caveats:

* Need to get rid of `on_initialize()` once for all, otherwise weird bugs occur with auto-starting timers in the notebook (I'll raise an issue)
* Doing `canvas.show()` several times with the same `Canvas` instance does not work. Users need to recreate the canvas explicitely. This needs to be clear in the documentation. Solving this would require a GLIR cache: possible in the future, but pretty low priority. The current behavior is good enough in my opinion (basically we have to make sure to create a new canvas in the same cell that we show it).

Tackle some of #607 (not all)
Close #615 (no need for a GLIR cache for now)

### CODE REVIEW

**Comment by rossant** [comment]
Ready for review/merge once Travis is happy.

**Comment by rossant** [comment]
AppVeyor fails, not sure why..

**Comment by Eric89GXL** [comment]
Can you rebase against latest `master`? That should fix AppVeyor.

**Comment by rossant** [comment]
Just did it, nothing happened.

**Comment by Eric89GXL** [comment]
Oh -- if you merge #695 then rebase it should work :)

**Comment by rossant** [comment]
OK done :)

File: vispy/app/backends/_ipynb_webgl.py [hunk 1]
```diff
@@ -312,18 +312,30 @@ def _vispy_stop(self):
...
 # ---------------------------------------------------------- IPython Widget ---
+def _stop_timers(canvas):
+    """Stop all timers in a canvas."""
+    for attr in dir(canvas):
+        try:
+            attr_obj = getattr(canvas, attr)
+        except NotImplementedError:
+            attr_obj = None
+        if isinstance(attr_obj, Timer):
+            attr_obj.stop()
```
Comment (Eric89GXL) [inline vispy/app/backends/_ipynb_webgl.py]: Can you mention why you needed to do try/except, and put it in the comments? If I came to this code to modify it later I would be confused...

File: vispy/html/static/js/webgl-backend.js [hunk 2]
```diff
@@ -1,15 +1,5 @@
 // Load Vispy.js in the notebook.
 IPython.load_extensions("vispy.min");
-IPython.load_extensions("jquery.mousewheel.min");
-
-// HACK: this is UGLY but I didn't find a better way to do it
```
Comment (Eric89GXL) [inline vispy/html/static/js/webgl-backend.js]: hooray, removing a HACK comment :)

**Comment by Eric89GXL** [comment]
Other than my minor gripe, LGTM +1 for merge. I agree the multiple `.show()` case is just a limitation we can tell users about, I don't think it's really a show-stopper. I'll let @almarklein or @lcampagn have a look before merge, though, since I'm not too familiar with this stuff.

File: vispy/app/backends/_ipynb_webgl.py (hunk 1 above)
Comment (rossant) [inline vispy/app/backends/_ipynb_webgl.py]: Done!

**Comment by rossant** [comment]
I had to rebase because of the example flake problems.

**Comment by rossant** [comment]
Everybody is happy hurray :)

**Comment by almarklein** [comment]
I don't fully understand every bit, but I could not detect anything bad happening :) LGTM

**Pull request merged**